calico_containers/tests/st/docker_host.py

```diff
@@ -15,16 +15,21 @@ def __init__(self, name):
         pwd = sh.pwd().stdout.rstrip()
         docker.run("--privileged", "-v", pwd+":/code", "--name", self.name, "-tid", "jpetazzo/dind")
```

Comment:
This isn't a build container, it's a system test container.  It would be really handy to have the actual ST container artifact as something you can pull down from failed ST runs.  The way this framework is set up, we repeat the work of caching the relevant container images for every host (mercifully we only pull from the external source once per run).

Furthermore, Docker's FS layering means that we should be able to engineer a path where we don't need to build from scratch every time there is a code change---we just re-layer the final changes.